### ISSUE TITLE
Move `dask-mpi` out of distributed codebase

### ISSUE THREAD

**Issue by kmpaul** [dask/dask-mpi#1]
@mrocklin Created this repository to move the `dask-mpi` code out of `distributed`.  Obviously, it should go here.

**Comment by kmpaul** [comment]
I have now copied the `distributed/cli/dask_mpi.py` and its test file `distributed/cli/tests/test_dask_mpi.py` into the `dask_mpi/cli/` directory, moving all of the associated commit history with those files.

**Comment by mrocklin** [comment]
I think it's fine to drop history here if it makes you more productive, and
just point to the commit in the dask/distributed repository where you
copied from.

I say this because I *think* that I'm mostly the person who set up dask-mpi
and because it's mostly pretty small.

**Comment by mrocklin** [comment]
Great.  Lets keep it.

**Comment by kmpaul** [comment]
**Update:**
I believe that the `master` branch of `dask-mpi` now has `dask_mpi/cli/dask_mpi.py` and its corresponding test `dask_mpi/cli/tests/test_dask_mpi.py` in place and working.  I've updated the `setup.py` script to include the `entry_point` for the `dask-mpi` command line tool.

Along the way, I noticed a problem related to OpenMPI on MacOS.  It turns out that declaring `mpi4py` as a dependency (*e.g.*, in the `environment.yml` file) installs `openmpi` by default.  On MacOS, the conda-installed `openmpi` has what I would consider a bug, which requires the `--oversubscribe` argument to `mpirun` in order to launch a parallel application, even with fewer processes than cores on the machine.  I believe this was probably never discovered in `distributed` because `mpich` used to be the default MPI installed by conda, not `openmpi`.  And `mpich` works out of the box, as expected.

This lead me to add `mpich` as a dependency to the development environment (`environment-dev.yml`).  With this modification, I was able to get 3 out of 4 tests in `dask_mpi/cli/tests/test_dask_mpi.py` to pass.  However, **the `test_bokeh` test fails for reasons I don't understand**.  The pytest errors are long, so they can be seen [here](https://file.io/VWOgl9) (until December 27).
